runescape-api/src/main/java/net/runelite/rs/api/RSClient.java

```diff
@@ -332,4 +332,7 @@
 	@Construct
 	@Override
 	RSIndexedSprite createIndexedSprite();
+
+	@Import("isFriended")
+	boolean isFriended(String playerName, boolean onlyOnline);
```

Comment:
Preferably this would be implemented as a client mixin + also cached. I think @UniquePassive is also working on making the core client able to have a fast isFriended() since evidently this function is responsible for some noticeable fps drop in busy areas.

Reply:
Well, I can rebase the PR later, or @UniquePassive can rebase his work if this would be merged sooner. I assume the method would be still same, so only change needed would be to remove this part from here, but the calls to it would remain valid, so no changes in the actual Plugin code will not be needed.